lxd/openfga/client.go

```diff
@@ -0,0 +1,121 @@
+package openfga
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/lxc/lxd/shared/logger"
+
+	openfga "github.com/openfga/go-sdk"
+	"github.com/openfga/go-sdk/credentials"
+)
+
+type Client struct {
+	apiClient   *openfga.APIClient
+	authModelID string
+}
+
+// NewClient returns a new Client struct containing an OpenFGA APIClient.
+func NewClient(apiURL string, apiToken string, storeID string) (*Client, error) {
+	u, err := url.Parse(apiURL)
```

Comment:
Should this potentially be an `*api.URL`? Never mind if called using the config value directly as this will be a string.

Reply:
I've seen that this is called directly with a value retrieved from the config table so better to leave as a string.